src/hooks/useReportIDs.tsx

```diff
@@ -0,0 +1,171 @@
+     * to SidebarLinksData, so this context doesn't have an
+     * access to currentReportID in that case.
+     *
+     * So this is a work around to have currentReportID available
+     * only in testing environment.
+     */
+    currentReportIDForTests,
+}: WithReportIDsContextProviderProps) {
+    const currentReportIDValue = useCurrentReportID();
+    const derivedCurrentReportID = currentReportIDForTests ?? currentReportIDValue?.currentReportID;
+    const {activeWorkspaceID} = useActiveWorkspace();
+    const {canUseViolations} = usePermissions();
+
+    const policyMemberAccountIDs = useMemo(() => getPolicyMembersByIdWithoutCurrentUser(policyMembers, activeWorkspaceID, getCurrentUserAccountID()), [activeWorkspaceID, policyMembers]);
+    const chatReportsKeys = useMemo(() => Object.keys(chatReports ?? {}), [chatReports]);
+    const reportIDsWithErrors = useMemo(
+        () =>
+            // eslint-disable-next-line you-dont-need-lodash-underscore/reduce
+            _.reduce(
+                chatReportsKeys,
+                (errorsMap, reportKey) => {
+                    const report = chatReports?.[reportKey] ?? null;
+                    const allReportsActions = allReportActions?.[reportKey.replace(ONYXKEYS.COLLECTION.REPORT, ONYXKEYS.COLLECTION.REPORT_ACTIONS)] ?? null;
+                    const errors = OptionsListUtils.getAllReportErrors(report, allReportsActions, allTransactions) || {};
+                    if (Object.values(errors).length === 0) {
+                        return errorsMap;
+                    }
+                    return {...errorsMap, [reportKey.replace(ONYXKEYS.COLLECTION.REPORT, '')]: errors};
+                },
+                {},
+            ),
+        [chatReportsKeys, allReportActions, allTransactions, chatReports],
+    );
+
+    const getOrderedReportIDs = useCallback(
+        (currentReportID?: string) =>
+            SidebarUtils.getOrderedReportIDs(
+                currentReportID ?? null,
+                chatReports,
+                betas ?? [],
+                policies,
+                priorityMode,
+                allReportActions,
+                transactionViolations,
+                activeWorkspaceID,
+                policyMemberAccountIDs,
+                reportIDsWithErrors,
+                canUseViolations,
+            ),
+        [chatReports, betas, policies, priorityMode, allReportActions, transactionViolations, activeWorkspaceID, policyMemberAccountIDs, reportIDsWithErrors, canUseViolations],
+    );
+
+    const orderedReportIDs = useMemo(() => getOrderedReportIDs(), [getOrderedReportIDs]);
+
+    // We need to make sure the current report is in the list of reports, but we do not want
+    // to have to re-generate the list every time the currentReportID changes. To do that
+    // we first generate the list as if there was no current report, then here we check if
+    // the current report is missing from the list, which should very rarely happen. In this
```

Comment:
Could we explain what is the rare circumstance this happens at?
```suggestion
    // the current report is missing from the list, which should very rarely happen. In this
```

Reply:
@mountiny This comment is moved from `SidebarLinksData`, so I don't have that much of a context what's the rare case.

Reply:
ok thanks

Reply:
probably when the user is using the focus mode, but the user deep links to a certain report isn't displayed on the focus mode?

Reply:
@hurali97 what happens in the following case:

the current report ID changes
-> the new current report ID is already included in the ordered report IDs
-> this won't trigger the re-calculation of the ordered report IDs
-> the old report ID is still included in the ordered list when it shouldn't be included when you actually recalculate the ordered list?